flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/HeapMapState.java

```diff
@@ -140,25 +141,25 @@ public boolean contains(UK userKey) {
 	@Override
 	public Iterable<Map.Entry<UK, UV>> entries() {
 		Map<UK, UV> userMap = stateTable.get(currentNamespace);
-		return userMap == null ? null : userMap.entrySet();
+		return userMap == null ? Collections.emptySet() : userMap.entrySet();
 	}
 
 	@Override
 	public Iterable<UK> keys() {
 		Map<UK, UV> userMap = stateTable.get(currentNamespace);
-		return userMap == null ? null : userMap.keySet();
+		return userMap == null ? Collections.emptySet() : userMap.keySet();
 	}
 
 	@Override
 	public Iterable<UV> values() {
 		Map<UK, UV> userMap = stateTable.get(currentNamespace);
-		return userMap == null ? null : userMap.values();
+		return userMap == null ? Collections.emptyList() : userMap.values();
```

Comment:
I think we could align the return result instead of some are `Collections.emptySet()` while some are `Collections.emptyList()`.
Actually, any kind of empty collection is acceptable. However, I don't think we should make these returned result different.